src/routes/tools/annual-averages/_data.js

```diff
@@ -55,154 +25,196 @@ const convertToAnnual = (values) => {
+export async function getObserved(config, params, method = "GET") {
+  try {
+    const seriesList = getObservedSeries(config);
+    const promises = seriesList.map((series) =>
+      fetchSeries({ series, params, method })
+    );
+    const data = await Promise.all(promises);
+    return data;
+  } catch (error) {
+    throw new Error(error.message);
+  }
 }
 
-export function flattenData(_data) {
-  return _data.reduce((acc, series) => {
-    const seriesValues = series.values.map((d) => {
-      return {
-        ...d,
-        year: +d.date.getFullYear(),
-        key: series.key,
-        label: series.label,
-      };
-    });
-    acc.push(...seriesValues);
-    return acc;
-  }, []);
+export async function getModels(config, params, method = "GET") {
+  try {
+    const seriesList = getModelSeries(config);
+    const promises = seriesList.map((series) =>
+      fetchSeries({ series, params, method })
+    );
+    const data = await Promise.all(promises);
+    return data;
+  } catch (error) {
+    throw new Error(error.message);
+  }
 }
 
-export function getDataByDate(_arr) {
-  return Array.from(
-    group(_arr, (d) => d.year),
-    ([year, values]) => {
-      const date = new Date(year, 11, 31);
-      const rows = values.map((d) => {
-        if (d.min) {
-          return {
-            key: d.key,
-            label: d.label,
-            value: [d["min"], d["max"]],
-          };
-        } else {
-          return {
-            key: d.key,
-            label: d.label,
-            value: d.value,
-          };
-        }
-      });
-      return { date, values: rows };
-    }
-  );
+export async function getEnvelope(config, params, method = "GET") {
+  try {
+    const seriesList = getEnsembleSeries(config);
+    const promises = seriesList.map((series) =>
+      fetchSeries({ series, params, method })
+    );
+    const data = await Promise.all(promises);
+    return data.map((d) => {
+      return { ...d, values: buildEnvelope(d.values) };
+    });
+  } catch (error) {
+    throw new Error(error.message);
+  }
 }
 
-export function formatDataForExport(_arr) {
-  return _arr.map((item) => {
-    const row = {};
-    row.year = item.date.getFullYear();
-    item.values.forEach((d) => {
-      if (Array.isArray(d.value)) {
-        row[`${d.label} Min`] = d.value[0];
-        row[`${d.label} Max`] = d.value[1];
-      } else {
-        row[d.label] = d.value;
-      }
-    });
-    return row;
-  });
+/**
+ * Creates the param object used to fetch data from API
+ * @param {object} location
+ * @param {object} boundary - obj representing a mapbox layer
+ * @param {boolean} imperial - represents units
+ * @return {object} params
+ * @return {string} method
+ */
+export function getQueryParams({ location, boundary, imperial = true }) {
+  const params = {};
+  let method;
+  switch (boundary.id) {
+    case "locagrid":
+      params.g = `Point(${location.center[0]} ${location.center[1]})`;
+      params.imperial = imperial;
```

Comment:
instead of repeating `params.imperial = imperial` in each of the `case` statements you could just put them in the `params` object from the start: 

```js
const params = {
  imperial,
}
```

maybe even with the `stat` value as well, though it looks like it's not included in the first `case "locagrid"` statement.

Reply:
Moved the imperial prop. The `stat` prop is not used with point geometry. I think it does some weird stuff on the backend if included. Fixed via https://github.com/berkeley-gif/caladapt-website-2021/commit/924a6f678229b2a6f69810306e3a4600e5ba7d06